packages/server/src/semester/semester.controller.ts

```diff
@@ -1,16 +1,138 @@
-import { SemesterPartial } from '@koh/common';
-import { Controller, Get } from '@nestjs/common';
+import { OrganizationRole, SemesterPartial } from '@koh/common';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  InternalServerErrorException,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
 import { SemesterModel } from './semester.entity';
-import { ApplicationConfigService } from '../config/application_config.service';
+import { JwtAuthGuard } from '../guards/jwt-auth.guard';
+import { EmailVerifiedGuard } from '../guards/email-verified.guard';
+import { OrganizationRolesGuard } from '../guards/organization-roles.guard';
+import { OrganizationGuard } from 'guards/organization.guard';
+import { OrganizationModel } from 'organization/organization.entity';
+import { Roles } from 'decorators/roles.decorator';
 
 @Controller('semesters')
 export class SemesterController {
-  constructor(private readonly appConfig: ApplicationConfigService) {}
+  @Get(':oid')
+  @UseGuards(JwtAuthGuard, EmailVerifiedGuard) // safe for anyone to fetch (needed for semester filtering in courses page)
+  async getSemesters(
+    @Param('oid', ParseIntPipe) organizationId: number,
+  ): Promise<SemesterPartial[]> {
+    const organization = await OrganizationModel.findOne({
+      where: { id: organizationId },
+    });
+
+    if (!organization) {
+      throw new BadRequestException('Organization not found');
+    }
+
+    const semesters = await SemesterModel.find({
+      where: { organizationId: organization.id },
+    });
```

Comment:
```suggestion
   try {
    await OrganizationModel.findOneOrFail({
      where: { id: organizationId },
    });
    } catch {
       throw new BadRequestException('Organization not found');
    }

    const semesters = await SemesterModel.find({
      where: { organizationId: organizationId },
    });
```

I like to use findOneOrFail since I like to think it helps performance since typeORM doesn't need to convert it into a returned object but in reality it probably still does but eh take of this comment what you will

Reply:
Oh, I was not aware there was an upper limit in the org config. Do you still think its relevant? I'm sure we would ideally want to keep history for potentially longer.

Reply:
I noticed this applies to some of my other routes in this controller too so I switched them up too. This is useful!